### DISCUSSION TITLE
Questions occurs in the EDK2 environment building process

### DISCUSSION THREAD

**Discussion by ZBCMars** [tianocore/edk2#3044] [Q&A]
Hi, Im a beginner in it and faced some problems when I tried to build the environment. 

My config:

- Ubuntu 20.04
- GCC 9.4.0
- nasm 2.14.02
- Python 3.8.10

And I followed this flow:

`git clone https://github.com/tianocore/edk2.git`
`cd edk2`
`git submodule update --init`
`make -C BaseTools`
`export EDK_TOOLS_PATH=$HOME/src/edk2/BaseTools`
`. edksetup.sh BaseTools`

I also modified 2 places in ～/src/edk2/Conf/target.txt:

- line 20: ACTIVE_PLATFORM       = MdeModulePkg/MdeModulePkg.dsc
- line 54: TOOL_CHAIN_TAG        = GCC5

Then I tried to run 'build' but I occured errors below:

`uild.py...
 : error 7000: Failed to execute command
	make tbuild [/home/uefibc/src/edk2/Build/MdeModule/DEBUG_GCC5/IA32/MdePkg/Library/BaseLib/BaseLib]

build.py...
 : error F002: Failed to build module
	/home/uefibc/src/edk2/MdePkg/Library/BaseLib/BaseLib.inf [IA32, GCC5, DEBUG]
`

Any ideas? Thanks.

**Comment by osteffenrh** [comment]
Hi,

please use the latest version of nasm. 2.14 is too old.

**Comment by lersi** [comment]
I ran into the same problem, in my case I upgraded ubuntu to version 22, so the nasm package was newer, and it fixed the problem.

try to checkout to older stable branch of EDK2 like `edk2-stable202005`

also checkout my project, that tries to make sense of edk's build system.

repo link: [edk2_project_template](https://github.com/lersi/edk2_project_template)
